org.lflang/src/org/lflang/generator/ReactionInstance.java

```diff
@@ -448,6 +448,57 @@ public String toString() {
         return getName() + " of " + parent.getFullName();
     }
 
+    /**
+     * Determine logical execution time for each reaction during compile
+     * time based on immediate downstream logical delays (after delays and actions)
+     * and label each reaction with the minimum of all such delays.
+     */
+    public TimeValue getLogicalExecutionTime() {
+        if (this.let != null) {
+            return this.let;
+        }
+
+        if (this.parent.isGeneratedDelay()) {
```

Comment:
Because when we get the action delay by `after` lf syntax, the delay exists in the 'Delay' reactor, which is the additional reactor for only delay. So, if the parent is in the generated delay condition, we thought it should return the `TimeValue.ZERO`.